applications/FluidDynamicsApplication/python_scripts/apply_outlet_process.py

```diff
@@ -128,15 +128,17 @@ def _ComputeOutletCharacteristicVelocity(self):
         for node in self.outlet_model_part.GetCommunicator().LocalMesh().Nodes:
             vnode = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY)
             outlet_avg_vel_norm += math.sqrt(vnode[0]*vnode[0] + vnode[1]*vnode[1] + vnode[2]*vnode[2])
-            outlet_avg_vel_norm /= len(self.outlet_model_part.GetCommunicator().LocalMesh().Nodes)
-
-            # Perform MPI communication
-            self.outlet_model_part.GetCommunicator().SumAll(outlet_avg_vel_norm)                 # Accumulate the average velocity in each processor
-            outlet_avg_vel_norm /= self.outlet_model_part.GetCommunicator().TotalProcesses()     # Compute the average between processors
-
-            # Store the average velocity in the ProcessInfo to be used in the outlet inflow prevention condition
-            min_outlet_avg_vel_norm = 1.0
-            if (outlet_avg_vel_norm >= min_outlet_avg_vel_norm):
-                self.outlet_model_part.GetRootModelPart().ProcessInfo[KratosFluid.CHARACTERISTIC_VELOCITY] = outlet_avg_vel_norm
-            else:
-                self.outlet_model_part.GetRootModelPart().ProcessInfo[KratosFluid.CHARACTERISTIC_VELOCITY] = min_outlet_avg_vel_norm
+
+        outlet_avg_vel_norm /= len(self.outlet_model_part.GetCommunicator().LocalMesh().Nodes)  # Partition average value
```

Comment:
this is not correct. You need to get the global sum, and the global number of nodes in the model part... NOT the local one. otherwise the result will not be the same of all the processors